### ISSUE TITLE
Compose ignores .dockerignore when containing exclusion

### ISSUE THREAD

**Issue by multimac** [docker/compose#3931]
Following on from this comment, https://github.com/docker/compose/issues/1607#issuecomment-190654346
I've created a repository demonstrating the issue, https://github.com/multimac/compose-dockerignore

When including a file already excluded by a prior rule in a .dockerignore file, Compose stops ignoring any other files.

**Expected Results**

```
ignore_1  | /context/
ignore_1  | /context/.dockerignore
ignore_1  | /context/docker-compose.yml
ignore_1  | /context/Dockerfile
ignore_1  | /context/folder
ignore_1  | /context/folder/included
```

**Actual Results** - (/context/folder/ignored shouldn't be there)

```
ignore_1  | /context/
ignore_1  | /context/.dockerignore
ignore_1  | /context/docker-compose.yml
ignore_1  | /context/Dockerfile
ignore_1  | /context/folder
ignore_1  | /context/folder/included
ignore_1  | /context/folder/ignored
```

**Versions**

```
Windows 10 Pro
Docker version 1.12.0, build 8eab29e
docker-compose version 1.8.0, build d988a55
```

**Comment by hiromaily** [comment]
+1

**Comment by enkoder** [comment]
Anyone have an update on this one? This bug is making it impossible to tune the context size of my monolithic repository.

**Comment by ralphie02** [comment]
+1

**Comment by JohnLozano** [comment]
+1 

Still seeing this issue with 
docker-compose 1.13.0, build 1719ceb8
docker-py version: 2.2.1

Docker build respects the .dockerignore just fine, compose ignores it entirely.

**Comment by aensidhe** [comment]
I have same issue on mac. 

```
Containers: 3
 Running: 0
 Paused: 0
 Stopped: 3
Images: 30
Server Version: 17.06.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 44
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.36-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3.855GiB
Name: moby
ID: LE4E:ZN7A:OKB5:G65M:Y25C:LDQ4:MNNF:E6JR:TJ7D:CX7T:32EN:SJY2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 18
 Goroutines: 30
 System Time: 2017-07-21T12:59:47.84242558Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by DannyBen** [comment]
So, as I see the milestone updates, I understand the fix still didn't make it to 1.15.0-rc1 ?
